vxquery-rest/src/main/java/org/apache/vxquery/rest/core/VXQuery.java

```diff
@@ -0,0 +1,454 @@
+                        Error.builder()
+                                .withCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code())
+                                .withMessage("Error occurred when reading results from hyracks for result ID : " + request.getResultId())
+                                .build());
+            }
+            return resultResponse;
+        } else {
+            return APIResponse.newErrorResponse(request.getRequestId(),
+                    Error.builder()
+                            .withCode(HttpResponseStatus.NOT_FOUND.code())
+                            .withMessage("No query found for result ID : " + request.getResultId())
+                            .build());
+        }
+    }
+
+    private void readResults(HyracksJobContext jobContext, QueryResultResponse resultResponse) throws Exception {
+        int nReaders = 1;
+
+        if (hyracksDataset == null) {
+            hyracksDataset = new HyracksDataset(hyracksClientConnection, jobContext.getFrameSize(), nReaders);
+        }
+
+        FrameManager resultDisplayFrameMgr = new FrameManager(jobContext.getFrameSize());
+        IFrame frame = new VSizeFrame(resultDisplayFrameMgr);
+        IHyracksDatasetReader reader = hyracksDataset.createReader(jobContext.getJobId(), jobContext.getResultSetId());
+        IFrameTupleAccessor frameTupleAccessor = new ResultFrameTupleAccessor();
+
+        OutputStream resultStream = new ByteArrayOutputStream();
+
+        try (PrintWriter writer = new PrintWriter(resultStream, true)) {
+            while (reader.read(frame) > 0) {
+                writer.print(ResultUtils.getStringFromBuffer(frame.getBuffer(), frameTupleAccessor));
+                writer.flush();
+                frame.getBuffer().clear();
+            }
+        }
+
+        // TODO: 7/12/17 Set metrics
+        hyracksClientConnection.waitForCompletion(jobContext.getJobId());
+        LOGGER.log(Level.INFO, String.format("Result for resultId %d completed", jobContext.getResultSetId().getId()));
+        resultResponse.setResults(resultStream.toString());
+    }
+
+    /**
+     * Create a unique result set id to get the correct query back from the cluster.
+     *
+     * @return Result Set id generated with current system time.
+     */
+    protected ResultSetId createResultSetId() {
+        long resultSetId = atomicLong.incrementAndGet();
+        LOGGER.log(Level.FINE, String.format("Creating result set with ID : %d", resultSetId));
+        return new ResultSetId(resultSetId);
+    }
+
+    /**
+     * Start local virtual cluster with cluster controller node and node controller nodes. IP address provided for node
+     * controller is localhost. Unassigned ports 39000 and 39001 are used for client and cluster port respectively.
+     * Creates a new Hyracks connection with the IP address and client ports.
+     *
+     * @throws Exception
+     */
+    @SuppressWarnings("Duplicates")
+    public void startLocalHyracks() throws Exception {
```

Comment:
@prestoncarman Sure. Will move all of the configuration to vxquery-server once we agree on a design/place to aggregate those functions within vxquery-server. Thank you for pointing these things out.